api/client/client.go

```diff
@@ -200,12 +200,14 @@ func connect(ctx context.Context, cfg Config) (*Client, error) {
 			}
 
 			// Connect with dialer provided in creds.
-			if dialer, err := creds.Dialer(); err == nil {
+			if dialer, err := creds.Dialer(cfg.KeepAlivePeriod, cfg.DialTimeout); err == nil {
 				syncConnect(constants.APIDomain, &Client{
 					c:         cfg,
 					tlsConfig: tlsConfig,
 					dialer:    dialer,
 				})
+			} else if !trace.IsNotImplemented(err) {
+				sendError(trace.Wrap(err))
 			}
```

Comment:
This is generally my coding style so it'd be good to understand exactly why you prefer this suggestion for future changes.
1) Do you find it more readable to do `dialer, err := creds.Dialer(cfg.KeepAlivePeriod, cfg.DialTimeout)` on its own line rather than in an `if`? I like to do `if` statements this way to reduce code size which generally makes things more readable for me.
2) If 1 is true, I would have changed it to:
```
dialer, err := creds.Dialer(cfg.KeepAlivePeriod, cfg.DialTimeout)
if err == nil {
	syncConnect(constants.APIDomain, &Client{
		c:         cfg,
		tlsConfig: tlsConfig,
		dialer:    dialer,
	})
} else if !trace.IsNotImplemented(err) {
	sendError(trace.Wrap(err))
}			
```
Do you prefer the `err != nil` check to always come first for code uniformity, or is there another reason?

Reply:
Sorry, should've been clearer.
It's about always handling the non-nil error first. Most readers of Go code will assume that the first indented block is error handling and could confuse `if err == nil` for `if err != nil`.

https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow